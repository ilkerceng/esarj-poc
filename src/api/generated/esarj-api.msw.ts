/**
 * Generated by orval v6.21.0 🍺
 * Do not edit manually.
 * Swagger Esarj
 * OpenAPI spec version: 1.0.0
 */
import { HttpResponse, delay, http } from 'msw';

export const getListUsersMock = () => [
  {
    id: 4124694674407424,
    personCompany: 'Kirk Hane',
    status: 1,
    accountType: 1,
    customerId: '4678140',
  },
  {
    id: 2358865563746304,
    personCompany: 'zero tolerance',
    status: 2,
    accountType: 2,
    customerId: '0005449',
  },
  {
    id: 245251658743808,
    personCompany: 'Gwendolyn Kshlerin',
    status: 1,
    accountType: 1,
    customerId: '0407619',
  },
  {
    id: 1852004477435904,
    personCompany: 'executive',
    status: 2,
    accountType: 2,
    customerId: '0635021',
  },
  {
    id: 7029827632627712,
    personCompany: 'Delbert Lynch',
    status: 1,
    accountType: 1,
    customerId: '9776395',
  },
  {
    id: 724463215706112,
    personCompany: 'foreground',
    status: 2,
    accountType: 2,
    customerId: '0037300',
  },
  {
    id: 3643498966286336,
    personCompany: 'Mitchell Hand',
    status: 1,
    accountType: 1,
    customerId: '0024670',
  },
  {
    id: 4919435817123840,
    personCompany: 'client-driven',
    status: 2,
    accountType: 2,
    customerId: '0003064',
  },
  {
    id: 3349953009156096,
    personCompany: 'Helen Borer',
    status: 1,
    accountType: 1,
    customerId: '0001724',
  },
  {
    id: 5470777105186816,
    personCompany: 'demand-driven',
    status: 2,
    accountType: 2,
    customerId: '0917125',
  },
];

export const getPostUserMock = () => ({
  id: '0079481',
  status: 1,
  accountType: 1,
  customerId: '0000210',
  userName: 'YIcMd3azl',
  firstName: 'Lauren',
  lastName: 'Stokes',
  email: 'Pierre6@yahoo.com',
  mobile: '275.944.7340 x1647',
});

export const getGetUserByIdMock = () => ({
  id: '0053569',
  status: 1,
  accountType: 1,
  customerId: '0093152',
  userName: 'MRNi5RzGdBmE51zq9GO',
  firstName: 'Marge',
  lastName: 'Turner',
  email: 'Elouise.Cartwright7@hotmail.com',
  mobile: '288.515.8297 x669',
});

export const getGetEnumsMock = () => ({
  accountTypes: [
    { id: 1, label: 'Personal' },
    { id: 2, label: 'Corporate' },
  ],
  statuses: [
    { id: 1, label: 'Active' },
    { id: 2, label: 'Passive' },
  ],
});

export const getSwaggerEsarjMock = () => [
  http.get('*/users', async () => {
    await delay(1000);
    return new HttpResponse(JSON.stringify(getListUsersMock()), {
      status: 200,
      headers: {
        'Content-Type': 'application/json',
      },
    });
  }),
  http.post('*/users', async () => {
    await delay(1000);
    return new HttpResponse(JSON.stringify(getPostUserMock()), {
      status: 200,
      headers: {
        'Content-Type': 'application/json',
      },
    });
  }),
  http.get('*/users/:userId', async () => {
    await delay(1000);
    return new HttpResponse(JSON.stringify(getGetUserByIdMock()), {
      status: 200,
      headers: {
        'Content-Type': 'application/json',
      },
    });
  }),
  http.get('*/enums', async () => {
    await delay(1000);
    return new HttpResponse(JSON.stringify(getGetEnumsMock()), {
      status: 200,
      headers: {
        'Content-Type': 'application/json',
      },
    });
  }),
];
