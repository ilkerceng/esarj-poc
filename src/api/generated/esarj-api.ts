/**
 * Generated by orval v6.21.0 üç∫
 * Do not edit manually.
 * Swagger Esarj
 * OpenAPI spec version: 1.0.0
 */
import { useMutation, useQuery } from '@tanstack/react-query';
import type {
  MutationFunction,
  QueryClient,
  QueryFunction,
  QueryKey,
  UseMutationOptions,
  UseQueryOptions,
  UseQueryResult,
} from '@tanstack/react-query';
import type {
  Enums,
  ListUsersParams,
  PostUserBody,
  User,
  UserListItem,
} from './model';
import { apiClient } from '../apiClient';
import type { ErrorType, BodyType } from '../apiClient';

// eslint-disable-next-line
type SecondParameter<T extends (...args: any) => any> = T extends (
  config: any,
  args: infer P,
) => any
  ? P
  : never;

/**
 * @summary Returns a list of users.
 */
export const listUsers = (
  params?: ListUsersParams,
  options?: SecondParameter<typeof apiClient>,
) => {
  return apiClient<UserListItem[]>(
    { url: `/users`, method: 'GET', params },
    options,
  );
};

export const getListUsersQueryKey = (params?: ListUsersParams) => {
  return [`/users`, ...(params ? [params] : [])] as const;
};

export const getListUsersQueryOptions = <
  TData = Awaited<ReturnType<typeof listUsers>>,
  TError = ErrorType<unknown>,
>(
  params?: ListUsersParams,
  options?: {
    query?: Partial<
      UseQueryOptions<Awaited<ReturnType<typeof listUsers>>, TError, TData>
    >;
    request?: SecondParameter<typeof apiClient>;
  },
) => {
  const { query: queryOptions, request: requestOptions } = options ?? {};

  const queryKey = queryOptions?.queryKey ?? getListUsersQueryKey(params);

  const queryFn: QueryFunction<Awaited<ReturnType<typeof listUsers>>> = () =>
    listUsers(params, requestOptions);

  return { queryKey, queryFn, ...queryOptions } as UseQueryOptions<
    Awaited<ReturnType<typeof listUsers>>,
    TError,
    TData
  > & { queryKey: QueryKey };
};

export type ListUsersQueryResult = NonNullable<
  Awaited<ReturnType<typeof listUsers>>
>;
export type ListUsersQueryError = ErrorType<unknown>;

/**
 * @summary Returns a list of users.
 */
export const useListUsers = <
  TData = Awaited<ReturnType<typeof listUsers>>,
  TError = ErrorType<unknown>,
>(
  params?: ListUsersParams,
  options?: {
    query?: Partial<
      UseQueryOptions<Awaited<ReturnType<typeof listUsers>>, TError, TData>
    >;
    request?: SecondParameter<typeof apiClient>;
  },
): UseQueryResult<TData, TError> & { queryKey: QueryKey } => {
  const queryOptions = getListUsersQueryOptions(params, options);

  const query = useQuery(queryOptions) as UseQueryResult<TData, TError> & {
    queryKey: QueryKey;
  };

  query.queryKey = queryOptions.queryKey;

  return query;
};

/**
 * @summary Returns a list of users.
 */
export const prefetchListUsers = async <
  TData = Awaited<ReturnType<typeof listUsers>>,
  TError = ErrorType<unknown>,
>(
  queryClient: QueryClient,
  params?: ListUsersParams,
  options?: {
    query?: Partial<
      UseQueryOptions<Awaited<ReturnType<typeof listUsers>>, TError, TData>
    >;
    request?: SecondParameter<typeof apiClient>;
  },
): Promise<QueryClient> => {
  const queryOptions = getListUsersQueryOptions(params, options);

  await queryClient.prefetchQuery(queryOptions);

  return queryClient;
};

/**
 * @summary Updates or Creates a user.
 */
export const postUser = (
  postUserBody: BodyType<PostUserBody>,
  options?: SecondParameter<typeof apiClient>,
) => {
  return apiClient<void>(
    {
      url: `/users`,
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      data: postUserBody,
    },
    options,
  );
};

export const getPostUserMutationOptions = <
  TError = ErrorType<unknown>,
  TContext = unknown,
>(options?: {
  mutation?: UseMutationOptions<
    Awaited<ReturnType<typeof postUser>>,
    TError,
    { data: BodyType<PostUserBody> },
    TContext
  >;
  request?: SecondParameter<typeof apiClient>;
}): UseMutationOptions<
  Awaited<ReturnType<typeof postUser>>,
  TError,
  { data: BodyType<PostUserBody> },
  TContext
> => {
  const { mutation: mutationOptions, request: requestOptions } = options ?? {};

  const mutationFn: MutationFunction<
    Awaited<ReturnType<typeof postUser>>,
    { data: BodyType<PostUserBody> }
  > = props => {
    const { data } = props ?? {};

    return postUser(data, requestOptions);
  };

  return { mutationFn, ...mutationOptions };
};

export type PostUserMutationResult = NonNullable<
  Awaited<ReturnType<typeof postUser>>
>;
export type PostUserMutationBody = BodyType<PostUserBody>;
export type PostUserMutationError = ErrorType<unknown>;

/**
 * @summary Updates or Creates a user.
 */
export const usePostUser = <
  TError = ErrorType<unknown>,
  TContext = unknown,
>(options?: {
  mutation?: UseMutationOptions<
    Awaited<ReturnType<typeof postUser>>,
    TError,
    { data: BodyType<PostUserBody> },
    TContext
  >;
  request?: SecondParameter<typeof apiClient>;
}) => {
  const mutationOptions = getPostUserMutationOptions(options);

  return useMutation(mutationOptions);
};

/**
 * @summary Returns a user by ID.
 */
export const getUserById = (
  userId: number,
  options?: SecondParameter<typeof apiClient>,
) => {
  return apiClient<User>({ url: `/users/${userId}`, method: 'GET' }, options);
};

export const getGetUserByIdQueryKey = (userId: number) => {
  return [`/users/${userId}`] as const;
};

export const getGetUserByIdQueryOptions = <
  TData = Awaited<ReturnType<typeof getUserById>>,
  TError = ErrorType<unknown>,
>(
  userId: number,
  options?: {
    query?: Partial<
      UseQueryOptions<Awaited<ReturnType<typeof getUserById>>, TError, TData>
    >;
    request?: SecondParameter<typeof apiClient>;
  },
) => {
  const { query: queryOptions, request: requestOptions } = options ?? {};

  const queryKey = queryOptions?.queryKey ?? getGetUserByIdQueryKey(userId);

  const queryFn: QueryFunction<Awaited<ReturnType<typeof getUserById>>> = () =>
    getUserById(userId, requestOptions);

  return {
    queryKey,
    queryFn,
    enabled: !!userId,
    ...queryOptions,
  } as UseQueryOptions<
    Awaited<ReturnType<typeof getUserById>>,
    TError,
    TData
  > & { queryKey: QueryKey };
};

export type GetUserByIdQueryResult = NonNullable<
  Awaited<ReturnType<typeof getUserById>>
>;
export type GetUserByIdQueryError = ErrorType<unknown>;

/**
 * @summary Returns a user by ID.
 */
export const useGetUserById = <
  TData = Awaited<ReturnType<typeof getUserById>>,
  TError = ErrorType<unknown>,
>(
  userId: number,
  options?: {
    query?: Partial<
      UseQueryOptions<Awaited<ReturnType<typeof getUserById>>, TError, TData>
    >;
    request?: SecondParameter<typeof apiClient>;
  },
): UseQueryResult<TData, TError> & { queryKey: QueryKey } => {
  const queryOptions = getGetUserByIdQueryOptions(userId, options);

  const query = useQuery(queryOptions) as UseQueryResult<TData, TError> & {
    queryKey: QueryKey;
  };

  query.queryKey = queryOptions.queryKey;

  return query;
};

/**
 * @summary Returns a user by ID.
 */
export const prefetchGetUserById = async <
  TData = Awaited<ReturnType<typeof getUserById>>,
  TError = ErrorType<unknown>,
>(
  queryClient: QueryClient,
  userId: number,
  options?: {
    query?: Partial<
      UseQueryOptions<Awaited<ReturnType<typeof getUserById>>, TError, TData>
    >;
    request?: SecondParameter<typeof apiClient>;
  },
): Promise<QueryClient> => {
  const queryOptions = getGetUserByIdQueryOptions(userId, options);

  await queryClient.prefetchQuery(queryOptions);

  return queryClient;
};

/**
 * @summary Returns enums.
 */
export const getEnums = (options?: SecondParameter<typeof apiClient>) => {
  return apiClient<Enums>({ url: `/enums`, method: 'GET' }, options);
};

export const getGetEnumsQueryKey = () => {
  return [`/enums`] as const;
};

export const getGetEnumsQueryOptions = <
  TData = Awaited<ReturnType<typeof getEnums>>,
  TError = ErrorType<unknown>,
>(options?: {
  query?: Partial<
    UseQueryOptions<Awaited<ReturnType<typeof getEnums>>, TError, TData>
  >;
  request?: SecondParameter<typeof apiClient>;
}) => {
  const { query: queryOptions, request: requestOptions } = options ?? {};

  const queryKey = queryOptions?.queryKey ?? getGetEnumsQueryKey();

  const queryFn: QueryFunction<Awaited<ReturnType<typeof getEnums>>> = () =>
    getEnums(requestOptions);

  return { queryKey, queryFn, ...queryOptions } as UseQueryOptions<
    Awaited<ReturnType<typeof getEnums>>,
    TError,
    TData
  > & { queryKey: QueryKey };
};

export type GetEnumsQueryResult = NonNullable<
  Awaited<ReturnType<typeof getEnums>>
>;
export type GetEnumsQueryError = ErrorType<unknown>;

/**
 * @summary Returns enums.
 */
export const useGetEnums = <
  TData = Awaited<ReturnType<typeof getEnums>>,
  TError = ErrorType<unknown>,
>(options?: {
  query?: Partial<
    UseQueryOptions<Awaited<ReturnType<typeof getEnums>>, TError, TData>
  >;
  request?: SecondParameter<typeof apiClient>;
}): UseQueryResult<TData, TError> & { queryKey: QueryKey } => {
  const queryOptions = getGetEnumsQueryOptions(options);

  const query = useQuery(queryOptions) as UseQueryResult<TData, TError> & {
    queryKey: QueryKey;
  };

  query.queryKey = queryOptions.queryKey;

  return query;
};

/**
 * @summary Returns enums.
 */
export const prefetchGetEnums = async <
  TData = Awaited<ReturnType<typeof getEnums>>,
  TError = ErrorType<unknown>,
>(
  queryClient: QueryClient,
  options?: {
    query?: Partial<
      UseQueryOptions<Awaited<ReturnType<typeof getEnums>>, TError, TData>
    >;
    request?: SecondParameter<typeof apiClient>;
  },
): Promise<QueryClient> => {
  const queryOptions = getGetEnumsQueryOptions(options);

  await queryClient.prefetchQuery(queryOptions);

  return queryClient;
};
