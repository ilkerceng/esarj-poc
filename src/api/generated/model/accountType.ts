/**
 * Generated by orval v6.21.0 üç∫
 * Do not edit manually.
 * Swagger Esarj
 * OpenAPI spec version: 1.0.0
 */

export type AccountType = (typeof AccountType)[keyof typeof AccountType];

// eslint-disable-next-line @typescript-eslint/no-redeclare
export const AccountType = {
  Personal: 'Personal',
  Corporate: 'Corporate',
} as const;
