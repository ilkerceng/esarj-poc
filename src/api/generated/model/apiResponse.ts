/**
 * Generated by orval v6.21.0 🍺
 * Do not edit manually.
 * Swagger Esarj
 * OpenAPI spec version: 1.0.0
 */

export interface ApiResponse {
  code?: number;
  message?: string;
  type?: string;
}
