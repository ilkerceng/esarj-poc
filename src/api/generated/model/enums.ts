/**
 * Generated by orval v6.21.0 üç∫
 * Do not edit manually.
 * Swagger Esarj
 * OpenAPI spec version: 1.0.0
 */
import type { AccountType } from './accountType';
import type { Status } from './status';

export interface Enums {
  accountTypes?: AccountType[];
  statuses?: Status[];
}
