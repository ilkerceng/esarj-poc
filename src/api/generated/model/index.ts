/**
 * Generated by orval v6.21.0 üç∫
 * Do not edit manually.
 * Swagger Esarj
 * OpenAPI spec version: 1.0.0
 */

export * from './accountType';
export * from './apiResponse';
export * from './enums';
export * from './listUsersParams';
export * from './postUserBody';
export * from './postUsersBody';
export * from './status';
export * from './user';
export * from './userListItem';
