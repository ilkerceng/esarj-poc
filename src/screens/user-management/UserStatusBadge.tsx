import { Badge, BadgeProps } from 'antd';
import { Status } from '../../api/generated/model';

type UserStatusBadgeProps = { status: Status } & Omit<BadgeProps, 'status'>;

export const UserStatusBadge = (props: UserStatusBadgeProps) => {
  return (
    <Badge
      status={props.status === Status.Active ? 'success' : 'error'}
      classNames={{ indicator: '!w-[15px] !h-[15px]', root: '!mr-2' }}
    />
  );
};
