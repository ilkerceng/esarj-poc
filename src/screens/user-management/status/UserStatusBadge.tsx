import { Badge, BadgeProps } from 'antd';
import { Status } from '../../../api/generated/model';

type UserStatusBadgeProps = {
  status: Status;
} & Omit<BadgeProps, 'status'>;

/**
 * Renders Badge according to User status Passive|Active.
 * @readonly
 * @param props
 * @example
 * <UserStatusBadge status={Status.Active} />
 */
export const UserStatusBadge = ({
  status,
  ...otherProps
}: UserStatusBadgeProps) => {
  return status ? (
    <Badge
      status={status === Status.Active ? 'success' : 'error'}
      classNames={{ indicator: '!w-[15px] !h-[15px]', root: '!mr-2' }}
      {...otherProps}
    />
  ) : null;
};
