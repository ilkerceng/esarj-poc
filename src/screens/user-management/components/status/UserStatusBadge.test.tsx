import { UserStatusBadge } from './UserStatusBadge';
import renderer from 'react-test-renderer';
import { Status } from '../../../../api/generated/model';

document.addEventListener('test', () => {});

test('Renders UserStatusBadge correctly', () => {
  const tree = renderer
    .create(<UserStatusBadge status={Status.Active} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// test('classNames set correctly', () => {
//   const { container } = render(<UserStatusBadge status={Status.Passive} />);
//   expect(container.firstChild).toHaveClass('');
// });
