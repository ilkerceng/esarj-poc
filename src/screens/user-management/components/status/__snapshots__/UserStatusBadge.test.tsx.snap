// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders UserStatusBadge correctly 1`] = `
<span
  className="ant-badge ant-badge-status ant-badge-not-a-wrapper !mr-2 css-dev-only-do-not-override-2q8sxy"
  style={{}}
>
  <span
    className="!w-[15px] !h-[15px] ant-badge-status-dot ant-badge-status-success"
    style={{}}
  />
</span>
`;
